clients/lsp-rust.el

```diff
@@ -1340,12 +1342,21 @@ meaning."
                    (pos (lsp--position-to-point position))
                    (overlay (make-overlay pos pos nil 'front-advance 'end-advance)))
              (let ((concat-label
-                    (string-join
-                     (mapcar
-                      (lambda (label)
-                        (when (lsp-structure-p label)
-                          (gethash "value" label "")))
-                      label))))
+                    (cl-typecase label
+                      (vector
+                       (string-join
+                        (mapcar
+                         (lambda (label)
+                           (when (lsp-structure-p label)
+                             (gethash "value" label "")))
```

Comment:
you should use `(lsp-get label :value)`

Reply:
This is needed in order to work fine with plists and hash-tables

Reply:
yeah, that was my first lisp code :-/